nossomercadolivre-api/src/main/java/com/github/nossomercadolivre/User.java

```diff
@@ -0,0 +1,126 @@
+package com.github.nossomercadolivre;
+
+import static javax.persistence.GenerationType.IDENTITY;
+
+import java.time.Instant;
+import java.util.Objects;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.validation.constraints.*;
+
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+@Entity
+@EntityListeners(AuditingEntityListener.class)
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = IDENTITY)
+    private Long id;
+
+    @CreatedDate
+    @Column(updatable = false, nullable = false)
+    private Instant createdDate;
+
+    @NotBlank
+    @Email
+    @Size(max = 100)
+    @Column(nullable = false, unique = true, length = 100)
+    private String email;
+
+    /**
+     * Password encrypted
+     */
+    @NotBlank
+    @Size(min = 6, max = 200)
+    @Column(nullable = false, length = 200)
+    private String password;
+
+    /**
+     * Just to frameworks
+     */
+    @Deprecated
+    private User() {
+    }
+
+    /**
+     *
+     * @param email email used to identify user
+     * @param newClearPassword password without any encryption
+     */
+    public User(String email, String newClearPassword) {
+        this.email = email;
+        this.password(newClearPassword);
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public Instant getCreatedDate() {
+        return createdDate;
```

Comment:
aqui vai retornar nulo se for chamado antes do save do repository.

Reply:
Tem razão. 
✅ Alterado